File: caper/samples/ruby/hello2.rb
require './hello2_parser.rb'

class SemanticAction
    def syntax_error
    end

    def stack_overflow
    end
    
    def downcast v
        v
    end

    def upcast v
        v
    end
    
    def Greet x, y
        print x.to_s + y.to_s + "\n"
        ""
    end
end

parser = Hello_world::Parser.new(SemanticAction.new)

parser.post :token_Hello, "Guten Tag, "
parser.post :token_World, "Welt"
parser.post :token_eof, ""

File: caper/samples/php/hello2.php
<?php

error_reporting(E_ALL);

require_once('hello2_parser.php');

class SemanticAction
{
    function syntax_error()
    {
    }

    function stack_overflow()
    {
    }
    
    function downcast($y)
    {
        return $y;
    }

    function upcast($y)
    {
        return $y;
    }
    
    function Greet($x, $y)
    {
        header('Content-Type: text/html');
        printf("%s%s\n", $x, $y);
        return "";
    }
}

$parser = new hello_world\Parser(new SemanticAction);

$parser->post(hello_world\Token::token_Hello, "Guten Tag, ");
$parser->post(hello_world\Token::token_World, "Welt");
$parser->post(hello_world\Token::token_eof, "");
